core/src/main/java/org/apache/iceberg/util/StructLikeWrapper.java

```diff
@@ -73,11 +73,6 @@ public boolean equals(Object other) {
       return false;
     }
 
-    int len = struct.size();
-    if (len != that.struct.size()) {
-      return false;
-    }
-
```

Comment:
Was this removed to ignore the extra columns coming from the file projection?

Reply:
Yes, it was. Since the wrapper's equality is based on the type, I think it doesn't make sense to check the size here.